instrumentation/http-url-connection/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/HttpUrlConnectionInstrumentation.java

```diff
@@ -122,16 +122,26 @@ public static void methodExit(
           // HttpURLConnection unnecessarily throws exception on error response.
           // None of the other http clients do this, so not recording the exception on the span
           // to be consistent with the telemetry for other http clients.
-          tracer().end(httpUrlState.context, new HttpUrlResponse(connection, responseCode));
+          instrumenter()
+              .end(
+                  httpUrlState.context,
+                  connection,
+                  new HttpUrlResponse(connection, responseCode),
+                  null);
         } else {
-          tracer().endExceptionally(httpUrlState.context, throwable);
+          instrumenter().end(httpUrlState.context, connection, null, throwable);
         }
         httpUrlState.finished = true;
       } else if (methodName.equals("getInputStream") && responseCode > 0) {
         // responseCode field is sometimes not populated.
         // We can't call getResponseCode() due to some unwanted side-effects
         // (e.g. breaks getOutputStream).
-        tracer().end(httpUrlState.context, new HttpUrlResponse(connection, responseCode));
+        instrumenter()
+            .end(
+                httpUrlState.context,
+                connection,
+                new HttpUrlResponse(connection, responseCode),
```

Comment:
Is the `connection` field of `HttpUrlResponse` used anymore? Could we just pass `responseCode` as the response instead?

Reply:
Nice catch!